packages/gatsby/src/services/extract-queries.ts

```diff
@@ -0,0 +1,20 @@
+import { IBuildContext } from "../state-machines/develop"
+
+import reporter from "gatsby-cli/lib/reporter"
+import { extractQueries as extractQueriesAndWatch } from "../query/query-watcher"
+import apiRunnerNode from "../utils/api-runner-node"
+
+export async function extractQueries({
+  parentSpan,
+}: IBuildContext): Promise<void> {
+  const activity = reporter.activityTimer(`onPreExtractQueries`, {
+    parentSpan,
+  })
+  activity.start()
+  await apiRunnerNode(`onPreExtractQueries`, {
+    parentSpan: activity.span,
+  })
+  activity.end()
+
+  await extractQueriesAndWatch({ parentSpan })
+}
```

Comment:
I assume this service would be called whenever we get into `extractingQueries` state - so it's not just first/initial one?

Showing activity (less important) and calling `onPreExtractQueries` each time we run this will change semantics of current API a bit (we only call it once - in extraction step we call explicitly during bootstrap)